prow/spyglass/gcsartifact_fetcher.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"context"
+	"log"
+	"path"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+// A fetcher for a GCS client
+type GCSArtifactFetcher struct {
+	client *storage.Client
+}
+
+// A location in GCS where prow job-specific artifacts are stored
+type GCSJobSource struct {
+	linkPrefix string
+	bucket     string
+	jobPath    string
+}
+
+// NewGCSArtifactFetcher creates a new ArtifactFetcher with a real GCS Client
+func NewGCSArtifactFetcher() *GCSArtifactFetcher {
+	c, err := storage.NewClient(context.Background(), option.WithoutAuthentication())
```

Comment:
Question: Should an authenticated client be used or must the bucket allow anymous access?
```
                gcsClient, err := storage.NewClient(ctx, option.WithCredentialsFile(o.GcsCredentialsFile))
```

Reply:
We currently require the bucket to be un-authed but should support hidden buckets in the future so that this is useful for private repos

Reply:
Please open an issue to track support for authed buckets.